pkg/controller/podgroup.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2023 The Kubeflow Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"sort"
+
+	"github.com/google/go-cmp/cmp"
+	common "github.com/kubeflow/common/pkg/apis/common/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	volcanov1beta1 "volcano.sh/apis/pkg/apis/scheduling/v1beta1"
+
+	kubeflow "github.com/kubeflow/mpi-operator/pkg/apis/kubeflow/v2beta1"
+)
+
+type replicaPriority struct {
+	priority int32
+
+	common.ReplicaSpec
+}
+
+// newPodGroup creates a new PodGroup for an MPIJob
+// resource. It also sets the appropriate OwnerReferences on the resource so
+// handleObject can discover the MPIJob resource that 'owns' it.
+func (c *MPIJobController) newPodGroup(mpiJob *kubeflow.MPIJob) *volcanov1beta1.PodGroup {
+	minMember := workerReplicas(mpiJob) + 1
+	queueName := mpiJob.Annotations[volcanov1beta1.QueueNameAnnotationKey]
+	var minResources *corev1.ResourceList
+	if schedulingPolicy := mpiJob.Spec.RunPolicy.SchedulingPolicy; schedulingPolicy != nil {
+		if schedulingPolicy.MinAvailable != nil {
+			minMember = *schedulingPolicy.MinAvailable
+		}
+		if len(schedulingPolicy.Queue) != 0 {
+			queueName = schedulingPolicy.Queue
+		}
+		if schedulingPolicy.MinResources != nil {
+			minResources = schedulingPolicy.MinResources
+		}
+	}
+	if minResources == nil {
+		minResources = c.calcPGMinResources(minMember, mpiJob.Spec.MPIReplicaSpecs)
```

Comment:
Yes. Previously, we could not use `minResources` since we don't calculate and set that:

https://github.com/kubeflow/mpi-operator/blob/c21942d1e27d196d06078f67f39a6f7a0e990dee/pkg/controller/mpi_job_controller.go#L1299-L1324

Reply:
If we configurable the `minResources`, scheduling part of Workers to Nodes is useful when the number of Workers is huge.